modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs

```diff
@@ -459,6 +464,11 @@ protected virtual async Task<IQueryable<IdentityUser>> GetFilteredQueryableAsync
             query = query.Where(identityUser => identityUser.Roles.Any(x => x.RoleId == roleId.Value) || identityUser.OrganizationUnits.Any(x => organizationUnitIds.Contains(x.OrganizationUnitId)));
         }
 
+        if (id.HasValue)
+        {
+            return query.Where(x => x.Id == id);
+        }
+
```

Comment:
Can you move to after `var query = await GetQueryableAsync();`

It has priority over `roleId.`